code/6502/README.md

```diff
@@ -0,0 +1,449 @@
+
+1. [An Overview of the 6502 Processor](#an-overview-of-the-6502-processor)
+1. [Common Instructions Explained](#common-instructions-explained)
+1. [Setting Up Your Development Environment](#setting-up-your-development-environment)
+1. [Resources](#resources)
+
+## An Overview of the 6502 processor
+
+Most assembly languages are unique to each kind of cpu as they are only one layer above machine code (which computers can understand) so instructions that it gives are only understandable by certain types of CPU. Hence, 6502 assembly language is used to program the 6502 processor (can you guess where the assembly language it got its name from?). The 6502 processor is an 8-bit-processor which means that it deals with data that are 8-bit numbers (see the section on binary number [here](/guide/math/number-systems.md#binary)). It can still process larger numbers but just does it 8-bit at a time. Eventhough a computer only understands [binary digits](/guide/math/number-systems.md#binary), when writing assembly language, we can use [hex numbers](/guide/math/number-systems.md#hexadecimal) which are a little easier to read. This is important because later when you look at code examples, you'll see that data is written with only 2 digit [hex numbers](/guide/math/number-systems.md#hexadecimal) like `$02`. Can you guess why it's only 2 digits? (hint: a binary number is base2 while a hexadecimal number is base16, answer [here](#number-systems))
+
+### Registers and RAM
+
+The 6502 processor has 6 [registers](/guide/cpu/instruction-cycle.md#fetch) (memory locations in the processor itself), three for general programming use (Register A, X, and Y) and three with specific purposes (program counter, stack pointer, and processor status). You (programmer) only have access to registers A, X, and Y and they can each only store one 8-bit number.
+
+Hence, we also have another source of memory: Random Access Memory (RAM). Data in the [RAM](/guide/cpu/instruction-cycle.md#fetch) can be accessed through 16-bit numbers called addresses. The 6502 processor is an 8-bit processor which means that all data is stored in 8-bits. You can imagine that both the RAM and Register only have two slots for a hexadecimal number (although in real life this would look more like 8 wires that can turn on and off with electricity, representing [binary digits](/guide/math/number-systems.md#binary)). 
+
+Hence, if the RAM or Register wants to store anything above 8-bits (represented up to a 2 digit hexadecimal number), it needs to use multiple memory locations. For example, memory addresses are 16-bit numbers which means that if we need to store data that points to a memory address, it is stored in two consecutive memory locations (eg. `$0102` and `$0103`) with the lower byte value stored first (since data is represented by numbers, it can be ordered and compared in terms of size). Although there's an exception: program counters store up to 16-bit numbers (or 4 digit hexademical numbers).
```

Comment:
```suggestion
So, if we want to store anything above 8 bits in a register or in memory, we need to use multiple locations. For example, memory addresses are 16-bits long. If we want to store a memory address, we would store it in two consecutive memory locations (eg. `$0102` and `$0103`). That being said, the program counter is an exception, as it can store 16-bit numbers.
```

Reply:
this is sooo much better, thank you!